ipaserver/plugins/agreement.py

```diff
@@ -0,0 +1,226 @@
+from ipalib import api, errors, output
+from ipalib import Bool, Str, StrEnum
+from ipalib.plugable import Registry
+from ipaserver.plugins.baseldap import (
+    LDAPObject, LDAPSearch, LDAPCreate, LDAPDelete,
+    LDAPUpdate, LDAPRetrieve, LDAPAddMember, LDAPRemoveMember,
+    global_output_params, pkey_to_value)
+from ipaserver.plugins.hbacrule import is_all
+from ipalib import _, ngettext
+from ipapython.dn import DN
+
+
+__doc__ = _("""
+""")
+
+register = Registry()
+
+
+@register()
+class agreement(LDAPObject):
+    """
+    User Agreement object.
+    """
+    container_dn = DN(("cn", "agreements"))
+    object_name = _('Agreement')
+    object_name_plural = _('Agreements')
+    object_class = ['ipaassociation', 'ipaagreements']
+    permission_filter_objectclasses = ['ipaagreements']
+    default_attributes = [
+        'cn', 'description', 'ipaenabledflag',
+        'ipacacategory', 'ipamemberca',
+        'ipacertprofilecategory', 'ipamembercertprofile',
+        'usercategory', 'memberuser',
+        'hostcategory', 'memberhost',
+        'servicecategory', 'memberservice',
+    ]
+    uuid_attribute = 'ipauniqueid'
+    rdn_attribute = 'ipauniqueid'
```

Comment:
Don't set rdn_attribute to make the primary_key ``cn`` the relative DN attribute. (so you get ``cn=My Agreement,cn=agreements,$SUFFIX`` instead of ``ipauniqueid=AAABBBBCCCDDD....,cn=agreements``).